platforms/core-configuration/base-diagnostics/src/integTest/groovy/org/gradle/api/tasks/diagnostics/TaskGraphIntegrationTest.groovy

```diff
@@ -0,0 +1,709 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks.diagnostics
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import spock.lang.Ignore
+import spock.lang.Issue
+
+class TaskGraphIntegrationTest extends AbstractIntegrationSpec {
+
+    def "shows simple graph of tasks"() {
+        given:
+        buildFile sampleGraph
+
+        when:
+        succeeds("root", "--task-graph")
+
+        then:
+        outputContains("""
+Tasks graph for: root
+\\--- :root (org.gradle.api.DefaultTask)
+     +--- :leaf1 (org.gradle.api.DefaultTask)
+     \\--- :middle (org.gradle.api.DefaultTask)
+          +--- :leaf1 (org.gradle.api.DefaultTask) (*)
+          \\--- :leaf2 (org.gradle.api.DefaultTask)
+
+(*) - details omitted (listed previously)
+""")
+        outputDoesNotContain("I'm a task called")
```

Comment:
❔ Should we use fixtures that check whether tasks were executed in this and other tests?

Reply:
I don't think it's important here, TBH